base/multidimensional.jl

```diff
@@ -367,6 +367,14 @@ module IteratorsMD
         return valid, (start[1], I...)
     end
 
+    function showarg(io::IO, c::CartesianIndices, toplevel)
+        if toplevel
+            print(io, "CartesianIndices ", UnitRange.(c.indices))
+        else
+            print(io, "::", typeof(c))
```

Comment:
I don't think this is the right behavior.
```julia
julia> v = view(rand(3,5), 1:2, 1:3);

julia> Base.showarg(stdout, v, true);
view(::Array{Float64,2}, 1:2, 1:3) with eltype Float64
julia> Base.showarg(stdout, v, false);
view(::Array{Float64,2}, 1:2, 1:3)
```
Here perhaps you could print the actual `AbstractRange` type, or just don't even check the value of `toplevel`.